src/dimension_slice.c

```diff
@@ -146,9 +152,26 @@ static ScanTupleResult
 dimension_vec_tuple_found(TupleInfo *ti, void *data)
 {
 	DimensionVec **slices = data;
-	DimensionSlice *slice = dimension_slice_from_slot(ti->slot);
+	DimensionSlice *slice;
 
-	lock_result_ok_or_abort(ti, slice);
+	switch (ti->lockresult)
+	{
+		case TM_SelfModified:
+		case TM_Ok:
+			break;
+#if PG12_GE
+		case TM_Deleted:
+#endif
+		case TM_Updated:
+			/* Treat as not found */
+			return SCAN_CONTINUE;
+		default:
+			elog(ERROR, "unexpected tuple lock status: %d", ti->lockresult);
+			pg_unreachable();
+			break;
+	}
```

Comment:
I realize that it is a good idea to not fetch the slice util after the checks, but why cannot `lock_result_ok_or_abort` be used? It is just used in two places, and this is one of them.

Reply:
We should not abort if a slice is not found or deleted.

Reply:
In fact, I think we should probably try to get rid of `lock_result_ok_or_abort`. But I wasn't sure about the locking that happens on the read path. (Actually not sure why the read path should take any locks at all, in particular given the problems we've had with that.)